teamscale-maven-plugin/src/main/java/com/teamscale/tia/maven/TiaMojoBase.java

```diff
@@ -0,0 +1,290 @@
+	public String accessToken;
+
+	/**
+	 * You can optionally use this property to override the code commit to which the coverage will be uploaded. Format:
+	 * {@code BRANCH:UNIX_EPOCH_TIMESTAMP_IN_MILLISECONDS}
+	 * <p>
+	 * If no end commit is manually specified, the plugin will try to determine the currently checked out Git commit.
+	 */
+	@Parameter
+	public String endCommit;
+
+	/**
+	 * You can optionally specify which code should be included in the coverage instrumentation. Each pattern is applied
+	 * to the fully qualified class names of the profiled system. Use {@code *} to match any number characters and
+	 * {@code ?} to match any single character.
+	 * <p>
+	 * Classes that match any of the include patterns are included, unless any exclude pattern excludes them.
+	 */
+	@Parameter
+	public String[] includes;
+
+	/**
+	 * You can optionally specify which code should be excluded from the coverage instrumentation. Each pattern is
+	 * applied to the fully qualified class names of the profiled system. Use {@code *} to match any number characters
+	 * and {@code ?} to match any single character.
+	 * <p>
+	 * Classes that match any of the exclude patterns are excluded, even if they are included by an include pattern.
+	 */
+	@Parameter
+	public String[] excludes;
+
+	/**
+	 * In order to instrument the system under test, a Java agent must be attached to the JVM of the system. The JVM
+	 * command line arguments to achieve this are by default written to the property {@code argLine}, which is
+	 * automatically picked up by Surefire and Failsafe and applied to the JVMs these plugins start. You can override
+	 * the name of this property if you wish to manually apply the command line arguments yourself, e.g. if your system
+	 * under test is started by some other plugin like the Spring boot starter.
+	 */
+	@Parameter
+	public String propertyName;
+
+	/**
+	 * Port on which the Java agent listens for commands from this plugin.
+	 */
+	@Parameter(defaultValue = "12888")
+	public String agentPort;
+
+	/**
+	 * Optional additional arguments to send to the agent. Each argument must be of the form {@code KEY=VALUE}.
+	 */
+	@Parameter
+	public String[] additionalAgentOptions;
+
+	/**
+	 * Changes the log level of the agent to DEBUG.
+	 */
+	@Parameter(defaultValue = "false")
+	public boolean debugLogging;
+
+	/**
+	 * Map of resolved Maven artifacts. Provided automatically by Maven.
+	 */
+	@Parameter(property = "plugin.artifactMap", required = true, readonly = true)
+	public Map<String, Artifact> pluginArtifactMap;
+
+	/**
+	 * The project build directory (usually: {@code ./target}). Provided automatically by Maven.
+	 */
+	@Parameter(defaultValue = "${project.build.directory}")
+	public String projectBuildDir;
+
+	/**
+	 * The running Maven session. Provided automatically by Maven.
+	 */
+	@Parameter(defaultValue = "${session}")
+	public MavenSession session;
+
+	private Path targetDirectory;
+	private String resolvedEndCommit;
+
```

Comment:
Add a `<skip>` configuration option to any Maven plugin. This gives you more flexibility, e.g., you can configure it outside of a profile with `skip=true` and only set `skip=false` in a profile, making the POM easier to read as the main build sequence is not scattered across profiles. Property would be something like `tia.skip` or so.